frontend/actions.js

```diff
@@ -26,32 +31,71 @@ export default ({setState, getState}) => {
       optionalArgsObj
     )
 
-  const loadWalletFromMnemonic = async (state, mnemonic) => {
-    setState(loadingAction(state, 'Loading wallet data...'))
+  const loadWalletFromMnemonic = async (state) => {
+    try {
+      setState(Object.assign(loadingAction(state, 'Loading wallet data...'), {mn: undefined}))
+
+      wallet = await Cardano.CardanoWallet({
+        cryptoProvider: 'mnemonic',
+        mnemonicOrHdNodeString: state.mnemonic,
+        config: CARDANOLITE_CONFIG,
+      })
+
+      const walletIsLoaded = true
+      const ownAddresses = await wallet.getOwnAddresses()
+      const transactionHistory = await wallet.getHistory()
+      const balance = await wallet.getBalance()
+      const sendAmount = {fieldValue: ''}
+      const sendAddress = {fieldValue: ''}
+      const sendResponse = ''
+      setState({
+        walletIsLoaded,
+        ownAddresses,
+        balance,
+        sendAmount,
+        sendAddress,
+        sendResponse,
+        transactionHistory,
+        loading: false,
+        mnemonic: '',
+      })
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(e.toString())
+
+      setState({
+        mnemonicValidationError: {
+          code: 'WalletInitializationError',
+        },
+      })
+    }
+  }
+
+  const loadDemoWallet = (state) => {
+    setState({
+      mnemonic: 'civil void tool perfect avocado sweet immense fluid arrow aerobic boil flash',
+      mnemonicValidationError: undefined,
+    })
+  }
 
-    wallet = await Cardano.CardanoWallet({
-      cryptoProvider: 'mnemonic',
-      mnemonicOrHdNodeString: mnemonic,
-      config: CARDANOLITE_CONFIG,
+  const generateMnemonic = (state) => {
+    setState({
+      mnemonic: Cardano.generateMnemonic(),
+      mnemonicValidationError: undefined,
     })
+  }
 
-    const walletIsLoaded = true
-    const ownAddresses = await wallet.getOwnAddresses()
-    const transactionHistory = await wallet.getHistory()
-    const balance = await wallet.getBalance()
-    const sendAmount = {fieldValue: ''}
-    const sendAddress = {fieldValue: ''}
-    const sendResponse = ''
+  const updateMnemonic = (state, e) => {
```

Comment:
Instead of `e` I would use value directly. If we change input for something else, then `e.target.value` will not be valid anymore.

Reply:
I would keep it simple - it would be cleaner to pass directly the value to the action but I think that making custom events directly in the components isn't too nice either. I think it's one of the tradeoffs we have to deal with for using Preact